src/shell/dotnet/Shell/Fdc3/ResolverUi/Fdc3ResolverUi.xaml.cs

```diff
@@ -0,0 +1,135 @@
+﻿// Morgan Stanley makes this available to you under the Apache License,
+// Version 2.0 (the "License"). You may obtain a copy of the License at
+// 
+//      http://www.apache.org/licenses/LICENSE-2.0.
+// 
+// See the NOTICE file distributed with this work for additional information
+// regarding copyright ownership. Unless required by applicable law or agreed
+// to in writing, software distributed under the License is distributed on an
+// "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express
+// or implied. See the License for the specific language governing permissions
+// and limitations under the License.
+
+using System;
+using System.Collections.Generic;
+using System.ComponentModel;
+using System.Linq;
+using System.Threading;
+using System.Windows;
+using System.Windows.Controls;
+using System.Windows.Input;
+using Finos.Fdc3;
+using Microsoft.Extensions.Logging;
+using Microsoft.Extensions.Logging.Abstractions;
+using MorganStanley.ComposeUI.Fdc3.DesktopAgent;
+using MorganStanley.ComposeUI.Shell.Fdc3.ResolverUi.Pages;
+
+namespace MorganStanley.ComposeUI.Shell.Fdc3.ResolverUi;
+
+/// <summary>
+/// Interaction logic for Fdc3ResolverUi.xaml
+/// </summary>
+public partial class Fdc3ResolverUi : Window, IDisposable
+{
+    private ILogger<Fdc3ResolverUi> _logger;
+    private readonly Page _simpleResolverUiPage;
+    private readonly Size _simpleResolverUiSize = new(500, 400);
+    private readonly Page _advancedResolverUiPage;
+    private readonly Size _advancedResolverUiSize = new(800, 600);
+    private Page? _currentPage;
+    private readonly List<ResolverUiAppData> _appData = new();
+    private readonly CancellationTokenSource _userCancellationTokenSource;
+
+    public IAppMetadata? AppMetadata { get; internal set; }
+    internal CancellationToken UserCancellationToken => _userCancellationTokenSource.Token;
+    
+    public Fdc3ResolverUi(
+        IEnumerable<IAppMetadata> apps,
+        ILogger<Fdc3ResolverUi>? logger = null)
+    {
+        _userCancellationTokenSource = new();
+        _logger = logger ?? NullLogger<Fdc3ResolverUi>.Instance;
+
+        InitializeComponent();
+
+        foreach (var app in apps)
```

Comment:
I'd recommend making more consistent use of MVVM constructs instead of writing code-behind like this.